Tests/js/views/services/plugins/assets/ez-versionsplugin-tests.js

```diff
@@ -0,0 +1,311 @@
+
+                    this.navigateCalled = true;
+                }, this),
+
+                routeUri: Y.bind(function (routeName, routeParams) {
+                    Assert.areSame(
+                        'editContentVersion',
+                        routeName,
+                        "The route provided should be 'editContentVersion'"
+                    );
+                    Assert.areSame(
+                        this.contentId,
+                        routeParams.id,
+                        "The id provided to the route doesn't match"
+                    );
+                    Assert.areSame(
+                        this.initialLanguageCode,
+                        routeParams.languageCode,
+                        "The language code provided to the route doesn't match"
+                    );
+                    Assert.areSame(
+                        this.versionId,
+                        routeParams.versionId,
+                        "The versionsId provided to the route doesn't match"
+                    );
+
+                    return "truc";
+                }, this)
+            };
+            this.service = new Y.Base();
+            this.view = new Y.View();
+            this.view.set('loadingError', false);
+            this.view.addTarget(this.service);
+            this.service.set('capi', this.capi);
+            this.service.set('app', this.app);
+            this.content = new Mock();
+            this.versionNo = "42";
+            this.contentId = "/zinedine/zidane";
+            this.contentName = "zidane";
+            this.initialLanguageCode = "eng-US";
+            this.versionId = "/path/to/version/42";
+            this.navigateCalled = false;
+
+            this.plugin = new Y.eZ.Plugin.Versions({
+                host: this.service
+            });
+        },
+
+        tearDown: function () {
+            this.plugin.destroy();
+            this.view.destroy();
+            this.service.destroy();
+            delete this.plugin;
+            delete this.view;
+            delete this.service;
+        },
+
+        _setupContentMock: function (error) {
+            var version = new Mock();
+
+            Mock.expect(version, {
+                method: 'get',
+                args: [Mock.Value.String],
+                run: Y.bind(function (attr) {
+                    if ( attr === 'id' ) {
+                        return this.versionId;
+                    } else if ( attr === 'initialLanguageCode' ) {
+                        return this.initialLanguageCode;
+                    }
+                    Y.fail('Unexpected call to get("' + attr + '")');
+                }, this),
+            });
+
+            Mock.expect(this.content, {
+                method: 'createDraft',
+                args: [Mock.Value.Object, this.versionNo, Mock.Value.Function],
+                run: function (options,versionId, callback) {
+                    callback(error, version);
```

Comment:
in case of error, the second parameter of the callback, you should provide something that looks like a Response object (`{}` would be ok here)